### PR TITLE
Aileen fix to do

### PR DESCRIPTION
**Pull request by aprilmayjune135**
- server side validation on socket gateway (getUserProfile and all GameInvite related methods);
- UserProfileCard improve flow, un-listen to event in onUnMounted();
- GameInvite un-listen to event in onUnMounted();

### CODE REVIEW

File: frontend/src/components/GameInvite.vue [hunk 1]
```diff
@@ -52,28 +52,35 @@ const receivedInviteList = ref();
...
 
 setTimeout(() => {
-  socket.emit("getReceivedGameInvites", storeUser.state.user.id);
+  socket.emit("getReceivedGameInvites", { data: storeUser.state.user.id });
 }, 100);
 
-socket.on("getReceivedGameInvites", (response) => {
-  receivedInviteList.value = response;
-});
+onMounted(() => {
```
Comment (Mollie-S) [inline frontend/src/components/GameInvite.vue:58]: Sometimes the game is not rendered.
The steps to reproduce:
1. Send the game invite to another user.
2. Join the queue.
3. The other user ignores your invitation and clicks 'Find a Match'

==>> the game starts but it doesn't render(sometimes it does for one player)

**Review by Mollie-S — CHANGES_REQUESTED** [review]

File: frontend/src/components/GameInvite.vue (hunk 1 above)
Comment (Mollie-S) [reply frontend/src/components/GameInvite.vue:58]: I've checked the previous version, it was stable

Comment (aprilmayjune135) [reply frontend/src/components/GameInvite.vue:58]: As discussed, this is due to the un-finished game flow, this will be updated in future PRs.

**Review by Mollie-S — APPROVED** [review]
It works. Can be merged.

**Pull request merged**